content/en/docs/devops-user-guide/how-to-use/pipeline-settings.md

```diff
@@ -1,9 +1,203 @@
+
+## Pipeline Settings
+
+### Basic Information
+
+![basic-info-tab](/images/docs/devops-user-guide/using-devops/pipeline-settings/basic-info-tab.png)
+
+On the **Basic Info** tab, you can customize the following information:
+
+- **Name**. The name of the pipeline. Pipelines in the same DevOps project must have different names.
+
+- **Project**. Projects will be grouped by their resources, which you can view and manage by project.
+
+- **Description**. The additional information to describe the pipeline. Description is limited to 256 characters.
+
+- **Code Repository (optional)**. You can select a code repository as the code source for the pipeline. In KubeSphere v3.1, you can select GitHub, GitLab, Bitbucket, Git, and SVN as the code source.
+
+  {{< tabs >}}
+
+  {{< tab "GitHub" >}}
+
+  ![code-source-github](/images/docs/devops-user-guide/using-devops/pipeline-settings/code-source-github.png)
+
+  If you select **GitHub**, you have to specify the token for accessing GitHub. You can click **Get Token** to generate a new token on GitHub. Click **Confirm** after entering the token and you can view your repository on the right. Click the **√** icon after you finish all operations.
+
+  {{</ tab >}}
+
+  {{< tab "GitLab" >}}
+
+  ![code-source-gitlab](/images/docs/devops-user-guide/using-devops/pipeline-settings/code-source-gitlab.png)
+
+  If you select **GitLab**, you have to specify the GitLab server, owner and repository name. You also need to specify a credential if it is needed for obtaining repository codes. Click the **√** icon after you finish all operations.
+
+  {{</ tab >}}
+
+  {{< tab "Bitbucket" >}}
+
+  ![code-source-bitbucket](/images/docs/devops-user-guide/using-devops/pipeline-settings/code-source-bitbucket.png)
+
+  If you select **Bitbucket**, you have to specify the Bitbucket server, username, and password. Click **Confirm** after entering the information. Click the **√** icon after you finish all operations.
+
+  {{</ tab >}}
+
+  {{< tab "Git" >}}
+
+  ![code-source-git](/images/docs/devops-user-guide/using-devops/pipeline-settings/code-source-git.png)
+
+  If you select **Git**, you have to specify the repository URL. You need to specify a credential if it is needed for obtaining repository codes. You can also click **Create a credential** to create a new credential. Click the **√** icon after you finish all operations.
+
+  {{</ tab >}}
+
+  {{< tab "SVN" >}}
+
+  ![code-source-svn](/images/docs/devops-user-guide/using-devops/pipeline-settings/code-source-svn.png)
+
+  If you select **SVN**, you have to specify the repository URL and the credential. You can also specify the branch included and excluded based on your needs. Click the **√** icon after you finish all operations.
+
+  {{</ tab >}}
+
+  {{</ tabs >}}
+
+### Advanced Settings with A Code Repository Selected
+
+If you selected a code repository, you can customize the following configurations on the **Advanced Settings** tab:
+
+**Branches Settings**
```

Comment:
Branches Settings
->
Branch Settings

Change the UI text in the console repository.

Reply:
UI changed. Please take a look at [the PR](https://github.com/kubesphere/console/pull/1762/commits/114049c675aaa768c807d526e7a8793e10e9a0e1).